core/vehicle/adapter.go

```diff
@@ -103,3 +103,13 @@ func (v *adapter) SetPlanSoc(ts time.Time, soc int) error {
 
 	return nil
 }
+
+func (v *adapter) SetRepeatingPlanEntries(entries string) error {
```

Comment:
```suggestion
func (v *adapter) SetRepeatingPlanEntries(entries  `[]RepeatingPlan`) error {
```

Das hier sollte bereits den korrekten Typen haben. Dieser muss im entsprechenden HTTP Handler aus dem Body erzeugt werden.

Reply:
Zusätzlich brauchst du den entsprechenden Getter, der die Pläne wieder zurückliefert:

```go
func (v *adapter) GetRepeatingPlanEntries() ([]RepeatingPlan, error) {
```

Diesen kannst du dann in `publishVehicles()` verwenden um das Struct, das gepublisht wird um das entsprechende Feld zu erweitern.

Reply:
Vielen Dank für die super Anleitung!